frontend/app/src/components/helper/BlockchainAccountSelector.vue

```diff
@@ -0,0 +1,201 @@
+            </div>
+          </v-chip>
+          <div v-else>
+            <div class="pr-2">
+              <v-avatar left>
+                <crypto-icon
+                  width="24px"
+                  :symbol="data.item.chain"
+                ></crypto-icon>
+              </v-avatar>
+              <span class="font-weight-bold mr-1">{{ data.item.label }}</span>
+              <span>({{ data.item.address | truncateAddress }})</span>
+            </div>
+          </div>
+        </template>
+        <template #item="data">
+          <div
+            class="blockchain-account-selector__list__item d-flex justify-space-between flex-grow-1"
+          >
+            <div class="blockchain-account-selector__list__item__address-label">
+              <v-chip color="grey lighten-3" filter>
+                <v-avatar left>
+                  <crypto-icon
+                    width="24px"
+                    :symbol="data.item.chain"
+                  ></crypto-icon>
+                </v-avatar>
+                <span class="font-weight-bold mr-1">{{ data.item.label }}</span>
+                <span>({{ data.item.address | truncateAddress }})</span>
+              </v-chip>
+            </div>
+            <div class="blockchain-account-selector__list__item__tags">
+              <tag-icon
+                v-for="tag in data.item.tags"
+                :key="tag"
+                class="mr-1"
+                :tag="tags[tag]"
+              ></tag-icon>
+            </div>
+          </div>
+        </template>
+      </v-autocomplete>
+    </div>
+    <v-card-text>
+      Showing results across
+      {{
+        selectedAccountsArray && selectedAccountsArray.length > 0
+          ? selectedAccountsArray.length
+          : 'all'
+      }}
+      accounts.
+    </v-card-text>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+import { createNamespacedHelpers } from 'vuex';
+import CryptoIcon from '@/components/CryptoIcon.vue';
+import TagIcon from '@/components/tags/TagIcon.vue';
+
+import {
+  Account,
+  AccountData,
+  AccountDataMap,
+  Tags,
+  Blockchain
+} from '@/typing/types';
+
+export interface GeneralAccount extends AccountData {
+  chain: Blockchain;
+}
+
+const { mapState } = createNamespacedHelpers('session');
+const { mapState: mapBalanceState } = createNamespacedHelpers('balances');
+
+@Component({
+  components: { CryptoIcon, TagIcon },
+  computed: {
+    ...mapState(['tags']),
+    ...mapBalanceState(['ethAccounts', 'btcAccounts'])
+  }
+})
+export default class BlockchainAccountSelector extends Vue {
+  @Prop({ required: true, type: Array, default: [] })
+  addresses!: Account[];
+  @Prop({ required: false, type: Boolean, default: false })
+  multiple!: boolean;
+
+  ethAccounts!: AccountDataMap;
+  btcAccounts!: AccountDataMap;
+  tags!: Tags;
+  selectedAccounts: GeneralAccount[] | GeneralAccount | null = null;
```

Comment:
why do we have this exception here for one item? can't we make it so that the selection is always an array? if there is 1 item it should still be an array of 1 item and if no values are selected it should be an empty array.

Reply:
Correct. But, the vuetify autocomplete component changes the structure of the associated value/v-model based on if it has the `multiple` property or not. Because this component is made to allow for both multiple or single selections, the output should be the same, however I didn't find a way to force vuetify to cast the `v-model` of the autocomplete always into an array.

So instead what I did was watch the `selectedAccounts` and then make sure that we're always sending out an array (`selectedAccountsArray`).

If you know a better way let's fix.

Reply:
I will replace the Watcher with a getter and a setter instead. It seems like the best alternative approach.

Reply:
Just make sure the watcher dialog works as it should. 

Right now there is a bug also. After the success of addition/editing/deleting the frontend shows ALL watchers a user has set, even if it is for other vaults. That should not happen. It should only show watchers of the vault that is being edited at the moment.

I think that's because of something wrong here:

https://github.com/rotki/rotki/pull/1040/files#diff-c0fd9961ee29022279ad3ca505055f18R261